password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java

```diff
@@ -43,41 +43,45 @@
 public class SunUnixMD5CryptTest {
 
     @Test
-    public void testParseCryptStringWithoutRounds() throws NoSuchAlgorithmException, InvalidKeySpecException {
-        String cryptString = "$md5$zrdhpMlZ$$wBvMOEqbSjU.hu5T2VEP01";
-
-        // Get the spec by parsing the crypt string
-        SunUnixMD5CryptPassword password = (SunUnixMD5CryptPassword) ModularCrypt.decode(cryptString);
-        assertEquals(0, password.getIterationCount());
+    public void testParseCryptStringWithoutRounds() throws NoSuchAlgorithmException, InvalidKeySpecException { 
+        testParseCryptString("$md5$zrdhpMlZ$$wBvMOEqbSjU.hu5T2VEP01");
+    }
 
-        // Use the spec to build a new crypt string and compare it to the original
-        assertEquals(cryptString, ModularCrypt.encodeAsString(password));
+    @Test
+    public void testParseCryptStringWithRounds() throws NoSuchAlgorithmException, InvalidKeySpecException { 
+        testParseCryptString("$md5,rounds=1000$saltstring$$1wGsmnKgDGdu03LxKu0VI1");
     }
 
     @Test
-    public void testParseCryptStringWithRounds() throws NoSuchAlgorithmException, InvalidKeySpecException {
-        String cryptString = "$md5,rounds=1000$saltstring$$1wGsmnKgDGdu03LxKu0VI1";
+    public void testParseCryptStringWithBareSalt() throws NoSuchAlgorithmException, InvalidKeySpecException { 
+        testParseCryptString("$md5,rounds=1500$saltstring$F9DNxgHVXWaeLS9zUaWXd.");
+    }
 
-        // Get the spec by parsing the crypt string
-        SunUnixMD5CryptPassword password = (SunUnixMD5CryptPassword) ModularCrypt.decode(cryptString);
-        assertEquals(1_000, password.getIterationCount());
+    
+    public void testParseCryptString(String cryptString) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        int assertionValue = 0;
 
-        // Use the spec to build a new crypt string and compare it to the original
-        assertEquals(cryptString, ModularCrypt.encodeAsString(password));
-    }
+        if (cryptString.equals("$md5$zrdhpMlZ$$wBvMOEqbSjU.hu5T2VEP01")) {
```

Comment:
Instead of determining the iteration count based on the hardcoded crypt strings, we could pass the iteration count as a parameter to the `testParseCryptString` method.

Reply:
Hello @fjuma 

I have made the modifications. Could you please check the below commit?
https://github.com/wildfly-security/wildfly-elytron/pull/2060/commits/d0c2e26ccf95617bbd03020a0e1d0e8f6fdfe68a

Reply:
Hi @shreya-pramod, the changes look great, thanks! Just noticed that there are some CI failures though due to some trailing spaces on the lines mentioned below. Would you be able to fix those?

```
/home/runner/work/wildfly-elytron/wildfly-elytron/password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java:46: Line has trailing spaces. [RegexpSingleline]
Error:  /home/runner/work/wildfly-elytron/wildfly-elytron/password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java:51: Line has trailing spaces. [RegexpSingleline]
Error:  /home/runner/work/wildfly-elytron/wildfly-elytron/password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java:56: Line has trailing spaces. [RegexpSingleline]
Error:  /home/runner/work/wildfly-elytron/wildfly-elytron/password/impl/src/test/java/org/wildfly/security/password/impl/SunUnixMD5CryptTest.java:60: Line has trailing spaces. [RegexpSingleline]
```